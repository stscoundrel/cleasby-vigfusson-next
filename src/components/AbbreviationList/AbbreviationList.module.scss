.wrap {
  display: flex;
}

.abbreviation {
  flex: 1;
  position: relative;
  padding-right: 20px;

  &::after {
    content: 'â†’';
    position: absolute;
    right: 10px;
    top: 0;
  }
}
.explanation {
  flex: 2;
}


@media only screen and (min-width: 768px) {
  .explanation {
    flex: 3;
  }
}