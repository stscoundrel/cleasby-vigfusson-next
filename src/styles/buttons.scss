.button {
  display: block;
  background-color: $color-blue;
  border: 2px solid $color-blue;
  color: white;
  padding: 5px 15px;
  margin: 20px 0;
  min-width: 100px;
  text-align: center;

  &:hover {
    background-color: transparent;
    color: $color-blue;
    text-decoration: none;
  }

  &--secondary {
    background: transparent;
    color: white;
    border: 2px solid white;

    &:hover {
      background-color: white;
      color: $color-blue;
    }
  }

  &--no-margs {
    margin: 0 !important;
  }
}


@media only screen and (min-width: 768px) {
  .button {
    display: inline-block;
    margin: 15px 0;
  }
}