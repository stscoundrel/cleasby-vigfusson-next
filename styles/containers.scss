.container {
  width: 100%;
  padding-right: 16px;
  padding-left: 16px;
  margin-right: auto;
  margin-left: auto;
  max-width: calc(100% - 32px);
}

@media (min-width: 640px) {
  .container {
    max-width: 640px;
  }
}

@media (min-width: 768px) {
  .container {
    padding-right: 35px;
    padding-left: 35px;
  }
}

@media (min-width: 1024px) {
  .container {
    max-width: 800px;
  }
}

@media (min-width: 1280px) {
  .container {
    max-width: 1280px;
  }
}

@media (min-width: 1536px) {
  .container {
    max-width: 1536px;
  }
}