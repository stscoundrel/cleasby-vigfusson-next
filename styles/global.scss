@import '~reset-css';
@import 'variables';
@import 'typography';
@import 'containers';

body,
html {
  max-width: 100%;
  overflow-x: hidden;
  scroll-behavior: smooth;
}

.capitalize {
  text-transform: capitalize;
}