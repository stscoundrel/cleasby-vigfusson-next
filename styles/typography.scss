html,
body {
  font-size: $body-font-size-mobile;
  line-height: $body-line-height-mobile;
  letter-spacing: $body-letter-spacing-mobile;
  font-weight: $body-font-weight-mobile;
  color: $color-body;
  font-family: $font-body;
}

h1,
.h1,
h2,
.h2,
h3,
.h3,
h4,
.h4,
h5,
.h5,
h6,
.h6 {
  font-family: $font-heading;
}

h1,
.h1 {
  font-size: $h1-font-size-mobile;
  line-height: $h1-line-height-mobile;
  letter-spacing: $h1-letter-spacing-mobile;
  font-weight: $h1-font-weight-mobile;
  margin-bottom: 0.75rem;
}

h2,
.h2 {
  font-size: $h2-font-size-mobile;
  line-height: $h2-line-height-mobile;
  letter-spacing: $h2-letter-spacing-mobile;
  font-weight: $h2-font-weight-mobile;
  margin-bottom: 1rem;
}

h3,
.h3 {
  font-size: $h3-font-size-mobile;
  line-height: $h3-line-height-mobile;
  letter-spacing: $h3-letter-spacing-mobile;
  font-weight: $h3-font-weight-mobile;
}

h4,
.h4 {
  font-size: $h4-font-size-mobile;
  line-height: $h4-line-height-mobile;
  letter-spacing: $h4-letter-spacing-mobile;
  font-weight: $h4-font-weight-mobile;
}

h5,
.h5 {
  font-size: $h5-font-size-mobile;
  line-height: $h5-line-height-mobile;
  letter-spacing: $h5-letter-spacing-mobile;
  font-weight: $h5-font-weight-mobile;
}

h6,
.h6 {
  font-size: $h6-font-size-mobile;
  line-height: $h6-line-height-mobile;
  letter-spacing: $h6-letter-spacing-mobile;
  font-weight: $h6-font-weight-mobile;
}

p {
  margin-bottom: 1rem;
}

strong {
  font-weight: bold;
}

em {
  font-style: italic;
}

small {
  font-size: smaller;
}

a {
  color: $color-blue;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}

@media only screen and (min-width: 992px) {
  html,
  body {
    font-size: $body-font-size;
    line-height: $body-line-height;
    letter-spacing: $body-letter-spacing;
    font-weight: $body-font-weight;
  }

  h1,
  .h1 {
    font-size: $h1-font-size;
    line-height: $h1-line-height;
    letter-spacing: $h1-letter-spacing;
    font-weight: $h1-font-weight;
  }

  h2,
  .h2 {
    font-size: $h2-font-size;
    line-height: $h2-line-height;
    letter-spacing: $h2-letter-spacing;
    font-weight: $h2-font-weight;
  }

  h3,
  .h3 {
    font-size: $h3-font-size;
    line-height: $h3-line-height;
    letter-spacing: $h3-letter-spacing;
    font-weight: $h3-font-weight;
  }

  h4,
  .h4 {
    font-size: $h4-font-size;
    line-height: $h4-line-height;
    letter-spacing: $h4-letter-spacing;
    font-weight: $h4-font-weight;
  }

  h5,
  .h5 {
    font-size: $h5-font-size;
    line-height: $h5-line-height;
    letter-spacing: $h5-letter-spacing;
    font-weight: $h5-font-weight;
  }

  h6,
  .h6 {
    font-size: $h6-font-size;
    line-height: $h6-line-height;
    letter-spacing: $h6-letter-spacing;
    font-weight: $h6-font-weight;
  }
}