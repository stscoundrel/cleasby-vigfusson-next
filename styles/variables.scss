// *********
// Variables
// *********

// Colors
$color-body:       #000;
$color-blue:       #3b4f68;

/* #################################### */

// Fonts
$font-body:           Arial, sans-serif;
$font-heading:        Arial, sans-serif;

/* #################################### */

// Body styles mobile
$body-font-size-mobile:        16px;
$body-line-height-mobile:      1.5;
$body-letter-spacing-mobile:   normal;
$body-font-weight-mobile:      normal;

// H1 styles mobile
$h1-font-size-mobile:          2.5rem;
$h1-line-height-mobile:        1.5;
$h1-letter-spacing-mobile:     0;
$h1-font-weight-mobile:        bold;

// H2 styles mobile
$h2-font-size-mobile:          1.75rem;
$h2-line-height-mobile:        1.5;
$h2-letter-spacing-mobile:     0;
$h2-font-weight-mobile:        normal;

// H3 styles mobile
$h3-font-size-mobile:          1.5rem;
$h3-line-height-mobile:        1.5;
$h3-letter-spacing-mobile:     0;
$h3-font-weight-mobile:        normal;

// H4 styles mobile
$h4-font-size-mobile:          1.25rem;
$h4-line-height-mobile:        1.5;
$h4-letter-spacing-mobile:     0;
$h4-font-weight-mobile:        normal;

// H5 styles mobile
$h5-font-size-mobile:          1rem;
$h5-line-height-mobile:        1.5;
$h5-letter-spacing-mobile:     0;
$h5-font-weight-mobile:        bold;

// H6 styles mobile
$h6-font-size-mobile:          1rem;
$h6-line-height-mobile:        1;
$h6-letter-spacing-mobile:     0;
$h6-font-weight-mobile:        normal;


// Body styles desktop
$body-font-size:        16px;
$body-line-height:      1.5;
$body-letter-spacing:   normal;
$body-font-weight:      normal;

// H1 styles desktop
$h1-font-size:          2.5rem;
$h1-line-height:        1.5;
$h1-letter-spacing:     0;
$h1-font-weight:        bold;

// H2 styles desktop
$h2-font-size:          2rem;
$h2-line-height:        1.5;
$h2-letter-spacing:     0;
$h2-font-weight:        normal;

// H3 styles desktop
$h3-font-size:          1.75rem;
$h3-line-height:        1.5;
$h3-letter-spacing:     0;
$h3-font-weight:        normal;

// H4 styles desktop
$h4-font-size:          1.5rem;
$h4-line-height:        1.5;
$h4-letter-spacing:     0;
$h4-font-weight:        normal;

// H5 styles desktop
$h5-font-size:          1.25rem;
$h5-line-height:        1.5;
$h5-letter-spacing:     0;
$h5-font-weight:        normal;

// H6 styles desktop
$h6-font-size:          1rem;
$h6-line-height:        1;
$h6-letter-spacing:     0;
$h6-font-weight:        normal;