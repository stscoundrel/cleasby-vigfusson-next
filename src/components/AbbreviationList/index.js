import styles from './AbbreviationList.module.scss'

export default function AbbreviationList({ abbreviations }) {
  return (
    <>
      {abbreviations.map(({ abbreviation, explanation }) => (
        <dl className={styles.wrap} key={abbreviation}>
          <dt className={styles.abbreviation}>
            <strong>{abbreviation}</strong>
          </dt>
          <dd className={styles.explanation}>{explanation}</dd>
        </dl>
      ))}
    </>
  )
}
