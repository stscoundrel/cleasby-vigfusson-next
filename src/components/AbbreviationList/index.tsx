import { Abbreviation } from 'lib/services/abbreviations'
import styles from './AbbreviationList.module.scss'

interface AbbreviationListProps{
  abbreviations: Abbreviation[]
}

export default function AbbreviationList({ abbreviations }: AbbreviationListProps) {
  return (
    <>
      {abbreviations.map(({ abbreviation, explanation }) => (
        <dl className={styles.wrap} key={abbreviation}>
          <dt className={styles.abbreviation}>
            <strong>{abbreviation}</strong>
          </dt>
          <dd className={styles.explanation}>{explanation}</dd>
        </dl>
      ))}
    </>
  )
}
