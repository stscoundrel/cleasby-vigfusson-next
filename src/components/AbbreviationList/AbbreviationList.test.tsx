import React from 'react'
import renderer from 'react-test-renderer'
import AbbreviationList from './index'
import styles from './AbbreviationList.module.scss'

describe('AbbreviationList component', () => {
  const abbreviations = [
    {
      abbreviation: 'f.',
      explanation: 'feminine.',
    },
    {
      abbreviation: 'n.',
      explanation: 'neuter.',
    },
  ]

  test('Matches the snapshot', () => {
    const tree = renderer.create(
      <AbbreviationList abbreviations={abbreviations} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct amount of abbreviations', () => {
    const tree = renderer.create(<AbbreviationList abbreviations={abbreviations} />)
    const { root } = tree

    expect(root.findAllByProps({ className: styles.abbreviation }).length).toEqual(2)
  })

  test('Has expected abbreviation content', () => {
    const tree = renderer.create(<AbbreviationList abbreviations={abbreviations} />)

    expect(JSON.stringify(tree)).toContain('f.')
    expect(JSON.stringify(tree)).toContain('feminine.')
    expect(JSON.stringify(tree)).toContain('n.')
    expect(JSON.stringify(tree)).toContain('neuter.')
  })
})
