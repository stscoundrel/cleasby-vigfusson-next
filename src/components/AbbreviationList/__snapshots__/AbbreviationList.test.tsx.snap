// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbbreviationList component Matches the snapshot 1`] = `
[
  <dl
    className="wrap"
  >
    <dt
      className="abbreviation"
    >
      <strong>
        f.
      </strong>
    </dt>
    <dd
      className="explanation"
    >
      feminine.
    </dd>
  </dl>,
  <dl
    className="wrap"
  >
    <dt
      className="abbreviation"
    >
      <strong>
        n.
      </strong>
    </dt>
    <dd
      className="explanation"
    >
      neuter.
    </dd>
  </dl>,
]
`;
